plugins/ipam/dhcp/lease.go

```diff
@@ -72,29 +72,34 @@ func AcquireLease(clientID, netns, ifName string) (*DHCPLease, error) {
 	log.Printf("%v: acquiring lease", clientID)
 
 	l.wg.Add(1)
-	go ns.WithNetNSPath(netns, true, func(_ *os.File) (e error) {
-		defer l.wg.Done()
-
-		link, err := netlink.LinkByName(ifName)
-		if err != nil {
-			errCh <- fmt.Errorf("error looking up %q", ifName)
-			return
-		}
+	go func() {
+		err := ns.WithNetNSPath(netns, true, func(_ *os.File) (e error) {
```

Comment:
The way I'd see it happening would be `errCh <- ns.WithNetNSPath()` and for the closure to return any errors to allow the wrapper to propagate - but maybe you'retrying to do something slightly different?